gemfiles/Gemfile.rails-5.1.4

```diff
@@ -3,3 +3,4 @@ source "https://rubygems.org"
 gemspec path: ".."
 
 gem "rails", "5.1.4"
+gem "puma"
```

Comment:
[Rails 5.1 requires puma in system tests](https://github.com/rails/rails/commit/50f697664edf0d2deff22f3f1a1c8e01d54a74ca) and this caused our system test to fail.

![Screen Shot 2021-07-23 at 5 48 31 PM](https://user-images.githubusercontent.com/796573/126844782-94c54498-efa3-4b69-98cb-7a5c3e3ffad0.png)